feat(authelia): Update Authelia question tooltips and defaults

depasseg opened the pull request:
**Description**
<!--
Replaced some descriptions with text from:
https://www.authelia.com/configuration/session
https://www.authelia.com/configuration/security

Added default settings to align with https://truecharts.org/charts/enterprise/authelia/Setup-Guide/#setup-authelia
-->
⚒️ Fixes  # <!--(issue)-->

**⚙️ Type of change**

- [X] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [X] 👀 I have performed a self-review of my own code
- [X] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [X] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

depasseg committed:
Update questions.yaml

Replaced some descriptions with text from:
https://www.authelia.com/configuration/session
https://www.authelia.com/configuration/security

Added default settings to align with https://truecharts.org/charts/enterprise/authelia/Setup-Guide/#setup-authelia

Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Update Chart.yaml

Bump Chart Version

Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Merge branch 'master' into patch-1

depasseg committed:
Update values.yaml

Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Merge branch 'master' into patch-1

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -773,7 +773,7 @@ questions:
                     - variable: subject
                       label: "Subject"
-                      description: "defines the subject to apply authorizations to. This parameter is optional and matching any user if not provided"
+                      description: "This criteria matches identifying characteristics about the subject. Currently this is either user or groups the user belongs to. This allows you to effectively control exactly what each user is authorized to access or to specifically require two-factor authentication to specific users. Subjects are prefixed with either user: or group: to identify which part of ...
```
Please use multiline yaml (`key: |`) strings, don't have that long single lines.

depasseg replied:
I hope I did that correctly.  I wasn't sure if I used the right syntax or put the line breaks in the right spot.

stavros-k reviewed (changes requested):
.

depasseg committed:
Added multiline yaml for long descriptions

Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg reviewed (commented):
Updated multiline YAML to break up long descriptions

depasseg committed:
Merge branch 'master' into patch-1

depasseg committed:
Merge branch 'master' into patch-1

Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Merge branch 'master' into patch-1

Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg reviewed (commented):
Fixed long lines

depasseg commented:
@stavros-k Does this need anything else?

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -230,14 +230,20 @@ questions:
-          description: "The name of the session cookie."
+          description: |
+            "The name of the session cookie. By default this is set to authelia_session. 
+            It’s mostly useful to change this if you are doing development or running multiple instances of Authelia."
```
```suggestion
            The name of the session cookie. By default this is set to authelia_session. 
            It’s mostly useful to change this if you are doing development or running multiple instances of Authelia.
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -230,14 +230,20 @@ questions:
+            "You can read about the SameSite cookie in detail on the MDN. In short setting SameSite to Lax is generally 
+            the most desirable option for Authelia. None is not recommended unless you absolutely know what you’re doing 
+            and trust all the protected apps. Strict is not going to work in many use cases and we have not tested it in 
+            this state but it’s available as an option anyway."
```
```suggestion
            You can read about the SameSite cookie in detail on the MDN. In short setting SameSite to Lax is generally 
            the most desirable option for Authelia. None is not recommended unless you absolutely know what you’re doing 
            and trust all the protected apps. Strict is not going to work in many use cases and we have not tested it in 
            this state but it’s available as an option anyway.
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -248,21 +254,27 @@ questions:
-          description: "The time in seconds before the cookie expires and session is reset."
+          description: |
+            "The period of time before the cookie expires and the session is destroyed. This is overriden by 
+            remember_me_duration when the remember me box is checked."
```
```suggestion
            The period of time before the cookie expires and the session is destroyed. This is overriden by 
            remember_me_duration when the remember me box is checked.
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -248,21 +254,27 @@ questions:
-          description: "The inactivity time in seconds before the session is reset."
+          description: |
+            "The period of time the user can be inactive for until the session is destroyed when the remember me box is 
+            not checked or is otherwise disabled. Useful if you want long session timers but don’t want unused devices to be vulnerable."
```
```suggestion
            The period of time the user can be inactive for until the session is destroyed when the remember me box is 
            not checked or is otherwise disabled. Useful if you want long session timers but don’t want unused devices to be vulnerable.
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -248,21 +254,27 @@ questions:
-          description: "The remember me duration"
+          description: |
+            "The period of time before the cookie expires and the session is destroyed when the remember me box is checked, a user 
+            selecting this option negates the inactivity timeout. Setting this to -1 disables this feature entirely."
```
```suggestion
            The period of time before the cookie expires and the session is destroyed when the remember me box is checked, a user 
            selecting this option negates the inactivity timeout. Setting this to -1 disables this feature entirely.
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -283,22 +295,26 @@ questions:
-          description: "The time range during which the user can attempt login before being banned."
+          description: |
+            "The period of time analyzed for failed attempts. For example if you set max_retries to 3 and find_time to 
+            2m this means the user must have 3 failed logins in 2 minutes."
```
```suggestion
          description: |
            The period of time analyzed for failed attempts. For example if you set max_retries to 3 and find_time to 
            2m this means the user must have 3 failed logins in 2 minutes.
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -283,22 +295,26 @@ questions:
+          description: |
+            "The period of time the user is banned for after meeting the max_retries and find_time configuration. 
+            After this duration the account will be able to login again."
           schema:
```
```suggestion
            The period of time the user is banned for after meeting the max_retries and find_time configuration. 
            After this duration the account will be able to login again.
          schema:
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -283,22 +295,26 @@ questions:
     group: "App Configuration"
     label: "Authentication Backend Provider"
-    description: "sed for verifying user passwords and retrieve information such as email address and groups users belong to."
+    description: "Used for verifying user passwords and retrieve information such as email address and groups users belong to."
```
```suggestion
    description: | 
      Used for verifying user passwords and retrieve information such as email 
      address and groups users belong to.
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -688,10 +704,13 @@ questions:
+          description: |
+            "The default policy defines the policy applied if no rules section apply to the information known about the request. 
+            It is recommended that this is configured to deny for security reasons. Sites which you do not wish to secure at all 
+            with Authelia should not be configured in your reverse proxy to perform authentication with Authelia at all for performance reasons."
```
```suggestion
          description: |
            The default policy defines the policy applied if no rules section apply to the information known about the request. 
            It is recommended that this is configured to deny for security reasons. Sites which you do not wish to secure at all 
            with Authelia should not be configured in your reverse proxy to perform authentication with Authelia at all for performance reasons.
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -758,7 +777,9 @@ questions:
-                      description: "The policy to apply to resources. It must be either 'bypass', 'one_factor', 'two_factor' or 'deny'."
+                      description: |
+                        "The specific policy to apply to the selected rule. This is not criteria for a match, this is the 
+                        action to take when a match is made."
```
```suggestion
                      description: |
                        The specific policy to apply to the selected rule. This is not criteria for a match, this is the 
                        action to take when a match is made.
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -773,7 +794,11 @@ questions:
+                        "This criteria matches identifying characteristics about the subject. Currently this is either 
+                        user or groups the user belongs to. This allows you to effectively control exactly what each user is 
+                        authorized to access or to specifically require two-factor authentication to specific users. Subjects 
+                        are prefixed with either user: or group: to identify which part of the identity to check."
```
```suggestion
                      description: |
                        This criteria matches identifying characteristics about the subject. Currently this is either 
                        user or groups the user belongs to. This allows you to effectively control exactly what each user is 
                        authorized to access or to specifically require two-factor authentication to specific users. Subjects 
                        are prefixed with either user: or group: to identify which part of the identity to check.
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -982,7 +1007,9 @@ questions:
-                                  description: "Configures the consent mode. This can be set to auto (default), explicit (consent required every time) or implicit (automatically assumes consent for every authorization, never asking the user if they wish to give consent.)"
+                                  description: |
+                                    "Configures the consent mode. This can be set to auto (default), explicit (consent required every time) or 
+                                    implicit (automatically assumes consent for every authorization, never asking the user if they wish to give consent.)"
```
```suggestion
                                  description: |
                                    Configures the consent mode. This can be set to auto (default), explicit (consent required every time) or 
                                    implicit (automatically assumes consent for every authorization, never asking the user if they wish to give consent.)
```

stavros-k committed:
Apply suggestions from code review

Signed-off-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

stavros-k commented on charts/enterprise/authelia/values.yaml:
```diff
@@ -267,21 +267,21 @@ authentication_backend:
-    user: CN=Authelia,DC=example,DC=com
-    plain_password: ""
+    user: uid=AutheliaSvcAcct,ou=people,DC=example,DC=com
+    plain_password: "Replace this with the AutheliaSvcAcct Password"
```
```suggestion
    plain_password: ""
```

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -433,40 +451,40 @@ questions:
                       schema:
                         type: string
-                        default: ""
+                        default: "Replace this with the AutheliaSvcAcct Password"
```
```suggestion
                        default: ""
```

stavros-k commented:
Have you validated that setting this default values does not trigger anything in the configmap/secret templates that users wouldn't expect to?

depasseg committed:
Fix typos

Co-authored-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>
Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Fix typos

Co-authored-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>
Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg commented:
> Have you validated that setting this default values does not trigger anything in the configmap/secret templates that users wouldn't expect to?

No I haven't.  I don't know how to do that.  I was just starting with changing the help text blocks from the upstream which in some cases had examples given.

depasseg committed:
Merge branch 'master' into patch-1

Signed-off-by: depasseg <depasseg@users.noreply.github.com>

xstar97 commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -433,34 +451,34 @@ questions:
                       schema:
                         type: string
-                        default: "CN=Authelia,DC=example,DC=com"
+                        default: "uid=AutheliaSvcAcct,ou=people,DC=example,DC=com"
```
```suggestion
                        default: "uid=admin,ou=people,DC=example,DC=com"
```

xstar97 commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -584,13 +602,13 @@ questions:
                       schema:
                         type: string
-                        default: "smtp.mail.svc.cluster.local"
+                        default: "smtp.gmail.com"
```
```suggestion
                        default: "smtp.mail.svc.cluster.local"
```

depasseg replied:
If we aren't going to use gmail as the example/default, then the documentation for https://truecharts.org/charts/enterprise/authelia/Setup-Guide/ should also be updated to remove gmail (and "steven" as the lldap admin for that matter)  

![image](https://github.com/truecharts/charts/assets/3201827/af46cb22-72b7-4653-92de-51f202315cc3)

stavros-k replied:
Steven was just the uid the guy that made that guide used.

But having real domains as placeholders is bad idea, there is a domain that anyone can safely use.
`example.com`

xstar97 commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -584,13 +602,13 @@ questions:
                       schema:
                         type: int
-                        default: 25
+                        default: 587
```
```suggestion
                        default: 25
```

xstar97 commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -782,7 +809,7 @@ questions:
                             schema:
                               type: string
-                              default: ""
+                              default: "group:lldap_password_manager"
```
```suggestion
                              default: ""
```

xstar97 commented on charts/enterprise/authelia/values.yaml:
```diff
@@ -267,20 +267,20 @@ authentication_backend:
 
     ## The username of the admin user.
-    user: CN=Authelia,DC=example,DC=com
+    user: uid=AutheliaSvcAcct,ou=people,DC=example,DC=com
```
```suggestion
    user: uid=admin,ou=people,DC=example,DC=com
```

xstar97 commented on charts/enterprise/authelia/values.yaml:
```diff
@@ -545,8 +545,8 @@ notifier:
-    host: smtp.mail.svc.cluster.local
-    port: 25
+    host: smtp.gmail.com
+    port: 587
```
```suggestion
    host: smtp.mail.svc.cluster.local
    port: 25
```

xstar97 reviewed (changes requested):
the following changes as follow.

the default user for lldap is admin; group: lldap_admin
due note that lldap does allow to create the custom admin user from its app config.

a user can create a custom user for the lldap_password_manager group just for password resets its not meant to be an "admin" group.

the smtp section should remained untouched and not directly set to a particular service....not many use gmail and would assume they need to set that instead of their preferred smtp service.

depasseg commented:
> the following changes as follow.
> 
> the default user for lldap is admin; group: lldap_admin due note that lldap does allow to create the custom admin user from its app config.
> 
> a user can create a custom user for the lldap_password_manager group just for password resets its not meant to be an "admin" group.
> 
> the smtp section should remained untouched and not directly set to a particular service....not many use gmail and would assume they need to set that instead of their preferred smtp service.

I pulled most of the settings, including the use of smtp.gmail.com from https://truecharts.org/charts/enterprise/authelia/Setup-Guide and https://truecharts.org/charts/stable/lldap/ 

I don't care what they are.  They should just be consistent.  Leaving things blank is confusing for new users trying to guess what syntax is required.

depasseg committed:
Update charts/enterprise/authelia/questions.yaml

Co-authored-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>
Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Update charts/enterprise/authelia/questions.yaml

Co-authored-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>
Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Update charts/enterprise/authelia/questions.yaml

seems odd to have the default different than the guide, but OK.

Co-authored-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>
Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Update charts/enterprise/authelia/questions.yaml

Co-authored-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>
Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Update charts/enterprise/authelia/values.yaml

Co-authored-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>
Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Update charts/enterprise/authelia/values.yaml

nowhere is setting up smtp.mail.svc.cluster.local explained.   I hope that works.

Co-authored-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>
Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg committed:
Merge branch 'master' into patch-1

Signed-off-by: depasseg <depasseg@users.noreply.github.com>

depasseg commented:
I'm not sure what to do with this, so I'll leave it to you to either delete or otherwise disposition.

stavros-k commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -433,34 +451,34 @@ questions:
                       schema:
                         type: string
-                        default: "CN=Authelia,DC=example,DC=com"
+                        default: "uid=admin,ou=people,DC=example,DC=com"
```
I'm not an LDAP guy, but above we have OU in caps, but here not. Does it matter?

Crow-Control replied:
doesn't really mater afaik.

Crow-Control commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -346,7 +364,7 @@ questions:
                       schema:
                         type: string
-                        default: "ldap://openldap.default.svc.cluster.local"
+                        default: "ldap://lldap-ldap.ix-lldap.svc.cluster.local:3890"
```
```suggestion
                        default: "ldap://openldap.default.svc.cluster.local"
```

VETO'ed, the original was a more clear example, the cluster-fucky 3x lldap is making it less clear, note more

Crow-Control commented on charts/enterprise/authelia/values.yaml:
```diff
@@ -197,7 +197,7 @@ authentication_backend:
     ## The url to the ldap server. Format: <scheme>://<address>[:<port>].
     ## Scheme can be ldap or ldaps in the format (port optional).
-    url: ldap://openldap.default.svc.cluster.local
+    url: ldap://lldap-ldap.ix-lldap.svc.cluster.local:3890
```
```suggestion
    url: ldap://openldap.default.svc.cluster.local
```
Also VETO'ed

Crow-Control commented on charts/enterprise/authelia/values.yaml:
```diff
@@ -249,7 +249,7 @@ authentication_backend:
     ## To allow sign in both with username and email, one can use a filter like
     ## (&(|({username_attribute}={input})({mail_attribute}={input}))(objectClass=person))
-    users_filter: ""
+    users_filter: "(&(|({username_attribute}={input})({mail_attribute}={input}))(objectClass=person))"
```
```suggestion
    users_filter: ""
```
No default needed in values.yaml at all. See notes

Crow-Control commented on charts/enterprise/authelia/values.yaml:
```diff
@@ -267,20 +267,20 @@ authentication_backend:
     ## If your groups use the `groupOfUniqueNames` structure use this instead:
     ##    (&(uniquemember={dn})(objectclass=groupOfUniqueNames))
-    groups_filter: ""
+    groups_filter: "(member={dn})"
```
```suggestion
    groups_filter: ""
```
Also no default needed in values.yaml see notes

Crow-Control commented on charts/enterprise/authelia/values.yaml:
```diff
@@ -267,20 +267,20 @@ authentication_backend:
 
     ## The username of the admin user.
-    user: CN=Authelia,DC=example,DC=com
+    user: uid=admin,ou=people,DC=example,DC=com
```
```suggestion
    user: CN=admin,DC=example,DC=com
```

Crow-Control commented on charts/enterprise/authelia/values.yaml:
```diff
@@ -228,10 +228,10 @@ authentication_backend:
 
     ## An additional dn to define the scope to all users.
-    additional_users_dn: OU=Users
+    additional_users_dn: OU=People
```
```suggestion
    additional_users_dn: OU=Users
```

This is only an lldap thing, not native openldap.

Crow-Control commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -433,34 +451,34 @@ questions:
                       schema:
                         type: string
-                        default: ""
+                        default: "(member={dn})"
```
```suggestion
                        default: ""
```

User and ldap backend dependent.

Crow-Control commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -405,21 +423,21 @@ questions:
                       schema:
                         type: string
-                        default: ""
+                        default: "(&(|({username_attribute}={input})({mail_attribute}={input}))(objectClass=person))"
```
```suggestion
                        default: ""
```
User and ldap backend dependend

Crow-Control commented on charts/enterprise/authelia/questions.yaml:
```diff
@@ -433,34 +451,34 @@ questions:
                       schema:
                         type: string
-                        default: "CN=Authelia,DC=example,DC=com"
+                        default: "uid=admin,ou=people,DC=example,DC=com"
```
```suggestion
                        default: "CN=admin,ou=people,DC=example,DC=com"
```

Commit:
Apply suggestions from code review

Signed-off-by: Kjeld Schouten <kjeld@schouten-lebbing.nl>

Commit:
Update Chart.yaml

Signed-off-by: Kjeld Schouten <kjeld@schouten-lebbing.nl>

Commit:
Update charts/enterprise/authelia/questions.yaml

Signed-off-by: Kjeld Schouten <kjeld@schouten-lebbing.nl>

Commit:
Merge branch 'master' into patch-1

Signed-off-by: Kjeld Schouten <kjeld@schouten-lebbing.nl>

Crow-Control commented:
@depasseg Please next PR, ensure you dont change existing defaults just because they work better with your *personal* setup.
@stavros-k Already removed a few and I had to clean-up the ldap ones. This needlessly takes maintainer time.

I've now cleaned it up and it should merge when passing tests.

depasseg commented:
> @depasseg Please next PR, ensure you dont change existing defaults just because they work better with your _personal_ setup. @stavros-k Already removed a few and I had to clean-up the ldap ones. This needlessly takes maintainer time.
> 
> I've now cleaned it up and it should merge when passing tests.

As I mentioned in my original PR, I updated the default values to align with the TrueCharts documentation:
https://truecharts.org/charts/enterprise/authelia/Setup-Guide/#setup-authelia
Do you want me to update the documentation as well to remove personalization?

And the tooltip description updates were based on:
https://www.authelia.com/configuration/session
https://www.authelia.com/configuration/security

Crow-Control commented:
> As I mentioned in my original PR, I updated the default values to align with the TrueCharts documentation: https://truecharts.org/charts/enterprise/authelia/Setup-Guide/#setup-authelia Do you want me to update the documentation as well to remove personalization?

It's already merged so obviously no.  Like I said: I already did so.

Just because we have a guide, does not mean defaults should follow that guide.
LLDAP is one of many backends and actually just a non-production backend at that.
